vignettes/custom_appearance.Rmd

```diff
@@ -90,3 +94,239 @@ lyt3 <- basic_table() %>%
 tbl3 <- build_table(lyt3, ex_adsl)
 tbl3
 ```
+
+### Top-left Materials
+
+The sequence of strings printed in the area between the column header display and the first row label \emph{top left material} can be modified during pre-processing within the table constructor `basic_table()` using `append_topleft`, the label position in row splits `split_rows_by` and during post-processing using the `top_left()` function. Note: No structure or indenting is applied to these strings.
+
+Within the table construction:
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("STRATA1") %>%
+  analyze("AGE") %>%
+  append_topleft("New top_left material here")
+
+build_table(lyt, DM)
+```
+
+Specify label position using the `split_rows` function. Notice the position of `STRATA1` and `SEX`.
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("STRATA1", label_pos = "topleft") %>%
+  split_rows_by("SEX", label_pos = "topleft") %>%
+  analyze("AGE")
+
+build_table(lyt, DM)
+```
+
+Post-processing using the `top_left()` function:
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("SEX") %>%
+  analyze(c("AGE", "STRATA1"), function(x) {
+    if (is.numeric(x)) {
+      in_rows(
+        "mean" = rcell(mean(x)),
+        "sd" = rcell(sd(x)),
+        .formats = c("xx.x"), .aligns = "left"
+      )
+    } else if (is.factor(x)) {
+      rcell(length(unique(x)), align = "right")
+    } else {
+      stop("Unsupported type")
+    }
+  }, show_labels = "visible", na_str = "NE") %>%
+  build_table(ex_adsl)
+
+# Adding top-left material
+top_left(lyt) <- "New top-left material here"
+
+lyt
+```
+
+### Table Inset
+
+Table title, table body, referential footnotes and and main footers can be inset from the left alignment of the titles and provenance footer materials. This can be modified within the table constructor `basic_table()` or during post-processing with `table_inset()`.
+
+Using the table constructor:
+
+```{r}
+lyt <- basic_table(inset = 5) %>%
+  analyze("AGE")
+
+build_table(lyt, DM)
+```
+
+Using the post-processing function:
+
+```{r}
+lyt <- basic_table() %>%
+  analyze("AGE")
+
+# without inset
+tbl <- build_table(lyt, DM)
+tbl
+
+# with inset of 5 characters
+table_inset(tbl) <- 5
+tbl
+```
+
+Below is an example with a table produced for clinical data. Compare the inset of the table and main footer between the two tables.
```

Comment:
It is good that we are pointing this out but we should be even more explicit and document that table inset:

- **Does not** apply to any title materials (main title, subtitles, page titles)
- **Does apply** to top-left materials
- **Does apply** to referential footnotes
- **Does apply** to main footer materials
- **Does not apply** to provenance footer materials
- **Does not apply** to any horizontal divider

```
>     analysisfun <- function(x, ...) {
+         in_rows(row1 = 5,
+                 row2 = c(1, 2),
+                 .row_footnotes = list(row1 = "row 1 rfn"),
+                 .cell_footnotes = list(row2 = "row 2 cfn"))
+     }
>     lyt <- basic_table(title = "Title says Whaaaat", subtitles = "Oh, ok.",
+                    main_footer = "ha HA! Footer!", prov_footer = "provenaaaaance") %>%
+         split_cols_by("ARM") %>%
+         analyze("AGE", afun = analysisfun)
>     result <-  build_table(lyt, ex_adsl)
> result
Title says Whaaaat
Oh, ok.

——————————————————————————————————————————————————
           A: Drug X   B: Placebo   C: Combination
——————————————————————————————————————————————————
row1 {1}       5           5              5       
row2       1, 2 {2}     1, 2 {3}       1, 2 {4}   
——————————————————————————————————————————————————

{1} - row 1 rfn
{2} - row 2 cfn
{3} - row 2 cfn
{4} - row 2 cfn
——————————————————————————————————————————————————

ha HA! Footer!

provenaaaaance
> table_inset(result) <- 5
> result
Title says Whaaaat
Oh, ok.

     ——————————————————————————————————————————————————
                A: Drug X   B: Placebo   C: Combination
     ——————————————————————————————————————————————————
     row1 {1}       5           5              5       
     row2       1, 2 {2}     1, 2 {3}       1, 2 {4}   
     ——————————————————————————————————————————————————

     {1} - row 1 rfn
     {2} - row 2 cfn
     {3} - row 2 cfn
     {4} - row 2 cfn
     ——————————————————————————————————————————————————

     ha HA! Footer!

provenaaaaance
```